public/static/js/distros.js

```diff
@@ -214,6 +223,47 @@ mciModule.controller('DistrosCtrl', function ($scope, $window, $http, $location,
       });
   };
 
+  $scope.updateSettingsList = function() {
+    // don't save if there aren't any populated fields
+    for (let [key, val] of Object.entries($scope.activeDistro.settings)) {
+        if (key !== "region" && val !== "" && val !== undefined) {
+          $scope.activeDistro.provider_settings[$scope.currentIdx] = $scope.activeDistro.settings;
+          return;
+        }
+    }
+    // if we haven't updated it, then we should remove from the list for now
+    $scope.activeDistro.provider_settings.splice($scope.currentIdx);
```

Comment:
Ok, you added a button to clear settings so I think that solves this

Reply:
Update: this does remove it from the list if no non-region fields are populated. In the route we now overwrite the existing list with our update list rather than marshalling over it, so doing this works to delete a region we don't want to provide settings for.